lib/remap-lang/src/parser.rs

```diff
@@ -157,13 +160,23 @@ impl<'a> Parser<'a> {
 
     /// Converts the set of known "root" rules into boxed [`Expression`] trait
     /// objects.
-    fn pairs_to_expressions(&mut self, pairs: Pairs<R>) -> Result<Vec<Expr>> {
+    ///
+    /// Each root expression needs to be considered infallible, otherwise the
+    /// program won't compile.
+    fn root_pairs_to_expressions(&mut self, pairs: Pairs<R>) -> Result<Vec<Expr>> {
         let mut expressions = vec![];
```

Comment:
It would be nicer if we could parse the whole root as a block and then error if `block.is_fallible == true`. A program is essentially just a top level block. Currently there is very little duplicated code between here and block, but that may change in future as the language grows..

I appreciate that may be out of scope for this pr.